src/OpenTelemetry/Trace/Builder/TracerProviderBuilderBase.cs

```diff
@@ -264,7 +272,24 @@ protected TracerProvider Build()
 
             var serviceProvider = services.BuildServiceProvider();
 
-            return new TracerProviderSdk(serviceProvider, ownsServiceProvider: true);
+            var sdkOptions = serviceProvider.GetRequiredService<IOptions<SdkOptions>>().Value;
+            if (!sdkOptions.TracingEnabled)
+            {
+                serviceProvider.Dispose();
+
+                return new NoopTracerProvider();
```

Comment:
@rajkumar-rangaraj Your design was returning the `TracerProvider.Default` for this case. I had that, but decided to add a no-op instance instead. The reason being: User's code is probably calling dispose/shutdown on the instance returned from build invocation. Seemed risky to return them something shared which they might kill accidentally?